osu.Game.Rulesets.Osu/Objects/Drawables/Pieces/SliderBall.cs

```diff
@@ -37,11 +46,20 @@ public Color4 AccentColour
         public readonly Drawable FollowCircle;
         private Drawable drawableBall;
         private readonly DrawableSlider drawableSlider;
+        private bool headHit;
 
         public SliderBall(Slider slider, DrawableSlider drawableSlider = null)
         {
             this.drawableSlider = drawableSlider;
             this.slider = slider;
+            //We no longer care about whether or not the slider is being tracked by the correct key if the head circle was never hit.
+            if (drawableSlider != null)
+                drawableSlider.HeadCircle.OnNewResult += (o, result) =>
+                {
+                    if (result.Type != HitResult.Miss)
+                        headHit = true;
```

Comment:
i'm not sure what this is trying to achieve but it's definitely not right. hitting the head should have no effect on tracking.